azext_edge/edge/providers/rpsaas/adr/assets.py

```diff
@@ -116,386 +116,707 @@ def create(
             product_code=product_code,
             serial_number=serial_number,
             software_revision=software_revision,
-            dp_publishing_interval=dp_publishing_interval,
-            dp_sampling_interval=dp_sampling_interval,
-            dp_queue_size=dp_queue_size,
+            ds_publishing_interval=ds_publishing_interval,
+            ds_sampling_interval=ds_sampling_interval,
+            ds_queue_size=ds_queue_size,
             ev_publishing_interval=ev_publishing_interval,
             ev_sampling_interval=ev_sampling_interval,
             ev_queue_size=ev_queue_size,
         )
+        # discovered
+        if discovered:
+            self.ops = self.discovered_ops
+            properties.pop("enabled", None)
+            properties["version"] = 1
+            properties["discoveryId"] = "discoveryid1"
 
         asset_body = {
             "extendedLocation": extended_location,
-            "location": location,
+            "location": location or cluster_location,
             "properties": properties,
             "tags": tags,
         }
-        poller = self.resource_client.resources.begin_create_or_update(
+        return self.ops.begin_create_or_replace(
+            resource_group_name,
+            asset_name,
+            resource=asset_body
+        )
+
+    def delete(self, asset_name: str, resource_group_name: str):
+        self.show(
+            asset_name=asset_name,
             resource_group_name=resource_group_name,
-            resource_provider_namespace=self.provider_namespace,
-            parent_resource_path="",
-            resource_type=self.resource_type,
-            resource_name=asset_name,
-            api_version=self.api_version,
-            parameters=asset_body
+            check_cluster=True
         )
-        return poller
+        return self.update_ops.begin_delete(
+            resource_group_name,
+            asset_name,
+        )
+
+    def show(
+        self, asset_name: str, resource_group_name: str, check_cluster: bool = False
+    ) -> dict:
+        # TODO: re-add try except when discovered is exposed
+        # try:
+        asset = self.ops.get(
+            resource_group_name=resource_group_name, asset_name=asset_name
+        )
+        self.update_ops = self.ops
```

Comment:
Reviewing further, looks like non show/list operations assume self.update_ops to be set with the expected operation instance, and the show function is responsible for setting the attribute. It doesn't seem optimal to have a show resource method alter the state of the class where other actions depend on show being run first.

Reply:
This is a set up for discovered assets so the same update/list/etc methods can be reused since they share almost all properties.

Flow will be (given a name):
1. try to get the asset - set the update ops as the asset ops
2. if 1 fails, try to get the discovered asset - set the update ops as the discovered asset ops
3. do the update/etc with the update ops

Reply:
will update this as I implement/do things with discovered